### PR TITLE
[chore] prepare v0.109.0 release

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Review by crobert-1 — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
If CI passes on https://github.com/open-telemetry/opentelemetry-collector-releases/pull/656 it would be create to get it in this release.

**Review by songy23 — APPROVED** [review]

**Comment by songy23** [comment]
I think https://github.com/open-telemetry/opentelemetry-collector-releases/pull/656 will need to be merged to this PR

This is what I did in 106: https://github.com/open-telemetry/opentelemetry-collector-releases/pull/615#discussion_r1696170013 following https://github.com/open-telemetry/opentelemetry-collector-releases/pull/611#pullrequestreview-2205171630

File: distributions/otelcol-k8s/manifest.yaml [hunk 1]
```diff
@@ -2,73 +2,80 @@ dist:
...
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sobjectsreceiver v0.108.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kubeletstatsreceiver v0.108.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/opencensusreceiver v0.108.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/otelarrowreceiver v0.108.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver v0.108.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator v0.108.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinreceiver v0.108.0
+  - gomod: go.opentelemetry.io/collector/receiver/otlpreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filelogreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/fluentforwardreceiver v0.109.0
```
Comment (songy23) [inline distributions/otelcol-k8s/manifest.yaml:54]:
```suggestion
  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/fluentforwardreceiver v0.109.0
  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/githubreceiver v0.109.0
```

cc @adrielp

Comment (codeboten) [reply distributions/otelcol-k8s/manifest.yaml:54]: the package is published and is alpha, i'm ok with committing the suggestion 👍🏻

Comment (adrielp) [reply distributions/otelcol-k8s/manifest.yaml:54]: appreciate you adding it in here @songy23 !

File: distributions/otelcol-k8s/manifest.yaml [hunk 2]
```diff
@@ -2,73 +2,81 @@ dist:
...
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/githubreceiver v0.109.0
```
Comment (TylerHelmuth) [inline distributions/otelcol-k8s/manifest.yaml:55]:
```suggestion
```
This component does not meet the [criteria](https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-k8s#rules-for-component-inclusion) for being in this distribution as it is not k8s-specific.

File: distributions/otelcol-contrib/manifest.yaml [hunk 3]
```diff
@@ -2,219 +2,226 @@ dist:
...
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/collectdreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchdbreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/datadogreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerstatsreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/elasticsearchreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/expvarreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filelogreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filestatsreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/flinkmetricsreceiver v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/fluentforwardreceiver v0.109.0
```
Comment (TylerHelmuth) [inline distributions/otelcol-contrib/manifest.yaml:144]:
```suggestion
  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/fluentforwardreceiver v0.109.0
  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/githubreceiver v0.109.0
```

**Review by TylerHelmuth — APPROVED** [review]

File: distributions/otelcol-k8s/manifest.yaml (hunk 2 above)
Comment (songy23) [reply distributions/otelcol-k8s/manifest.yaml]: Oops I meant to add it to contrib, thanks for catching this!

**Pull request merged**